core/src/main/scala/org/apache/spark/resource/ResourceProfileManager.scala

```diff
@@ -77,13 +85,27 @@ private[spark] class ResourceProfileManager(sparkConf: SparkConf,
     true
   }
 
+  def updateResourceReusePolicy(resourceNames: Set[String],
+    policy: ResourceProfileCompatiblePolicy): Unit = {
+    writeLock.lock()
+    try {
+      reuseResourceNames = resourceNames
+      reusePolicy = policy
```

Comment:
I added an extra param to specify policy for RDD.withResource

def withResources(rp: ResourceProfile,
    reuseResourceNames: Set[String], reusePolicy: ResourceProfileCompatiblePolicy): this.type = {

Reply:
> I'm not sure I follow this, I would expected the ExecutorAllocationManager.removeExecutors to be updated where it checks:
> 
> ```
>    } else if (newExecutorTotal - 1 < numExecutorsTargetPerResourceProfileId(rpId)) {
>            logDebug(s"Not removing idle executor $executorIdToBeRemoved because there " +
>             s"are only $newExecutorTotal executor(s) left (number of executor " +
>             s"target ${numExecutorsTargetPerResourceProfileId(rpId)})")
> ```
> 
> It should not remove an executor if it a task could run on it.

I didn't change removeExecutors logic, the executors are only removed when they timeout.